units/en/unit2/smolagents/why_use_smolagents.mdx

```diff
@@ -0,0 +1,69 @@
+![smolagents banner](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/smolagents/license_to_call.png)
+# Why use smolagents
+
+In this module, we will explore the pros and cons of using [smolagents](https://huggingface.co/docs/smolagents/en/index), helping you make an informed decision about whether it's the right framework for your needs.
+
+## What is `smolagents`?
+
+`smolagents` is a simple yet powerful framework for building AI agents. It provides LLMs with the _agency_ to interact with the real world, such as searching or generating images.
+
+As we learned in unit 1, AI agents are programs that use LLMs to generate **'thoughts'** based on **'observations'** to perform **'actions'**. Let's explore how this is implemented in smolagents.
+
+### Key Advantages of `smolagents`
+- **Simplicity:** Minimal code complexity and abstractions
+- **Flexible LLM Support:** Works with any LLM through integration with Hugging Face tools and external APIs
+- **Code-First Approach:** First-class support for Code Agents that directly write their actions in code
+- **HF Hub Integration:** Seamless integration with Hugging Face Hub, allowing the use of Gradio Spaces as tools
+
+### When to use smolagents?
+
+With these advantages in mind, when should we use smolagents over other frameworks? 
+
+Smolagents is ideal when:
+- You need a **lightweight and minimal solution.**
+- You want to **experiment quickly** without complex configurations.
+- Your **application logic is straightforward.**
+
+### Code vs. JSON Actions
+Unlike other frameworks where agents write actions in JSON that require parsing, `smolagents` **focuses on direct tool calls in code**, simplifying the execution process. 
+
+The following diagram illustrates this difference:
+
+![Code vs. JSON actions](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/code_vs_json_actions.png)
+
+If you don't remember the difference between Code vs JSON Actions, check [Unit 1, Section Actions](https://huggingface.co/learn/agents-course/unit1/actions#actions-enabling-the-agent-to-engage-with-its-environment).
+
+### Agent Types in `smolagents`
+
+Agents in `smolagents` operate as **multi-step agents**. 
+
+Each [`MultiStepAgent`](https://huggingface.co/docs/smolagents/main/en/reference/agents#smolagents.MultiStepAgent) performs:
+- One thought
+- One tool call and execution
+
+The framework offers two types of agents:
+1. **[CodeAgent](https://huggingface.co/docs/smolagents/main/en/reference/agents#smolagents.CodeAgent):** The default agent type that writes tool calls using Python
+2. **[ToolCallingAgent](https://huggingface.co/docs/smolagents/main/en/reference/agents#smolagents.ToolCallingAgent):** Writes tool calls in JSON
```

Comment:
```suggestion
In addition to using **[CodeAgent](https://huggingface.co/docs/smolagents/main/en/reference/agents#smolagents.CodeAgent)** as the primary type of agent, smolagents also supports **[ToolCallingAgent](https://huggingface.co/docs/smolagents/main/en/reference/agents#smolagents.ToolCallingAgent)**, which writes tool calls in JSON.
```

Felt a bit repetitive, and we stress code agent as the default / recommended way.